gson/src/main/java/com/google/gson/internal/bind/ReflectiveTypeAdapterFactory.java

```diff
@@ -215,6 +228,25 @@ protected BoundField(String name, boolean serialized, boolean deserialized) {
         in.beginObject();
         while (in.hasNext()) {
           String name = in.nextName();
+          if (this.defaultTyping && TYPE_FIELD_NAME.equals(name)) {
```

Comment:
```suggestion
          if (defaultTyping && TYPE_FIELD_NAME.equals(name)) {
```
Qualifying using `this` is not necessary